### DISCUSSION TITLE
No IP found

### DISCUSSION THREAD

**Discussion by EnricGarrigo** [jasonacox/tinytuya#290] [General]
"python -m tinytuya wizard" founds my 2 devices but can't found the devices IP. Then, I can't work locally.
![ERROR_IPs](https://user-images.githubusercontent.com/41141131/221914210-fb4371ca-c83f-4a67-97b8-df2c68d47c3f.PNG)

**Comment by uzlonewolf** [comment]
Is it a battery operated device?  If so, it will only connect and be visible on the network for a few seconds when it has something to report.  The rest of the time it will power down and sleep.

There are a few ideas discussed in https://github.com/jasonacox/tinytuya/issues/264 and https://github.com/jasonacox/tinytuya/issues/214 to get events from these devices, but none of them are particularly great.

**Comment by EnricGarrigo** [comment]
No is a battery device. Is a breaker switch. "AT-Q-SY1 WiFi/ ZigBee Din Rail"

In the 'devices.json' result of "python -m tinytuya wizard" there is none 'ip' key.

Any ideas? Thanks.

**Reply by uzlonewolf** [reply]
Are you connected to the same network as the devices, and are broadcast packets allowed on the network?  If not, you can try force-scanning for the devices with `python -m tinytuya scan -force 192.168.x.y/24`

**Comment by demonreeper** [comment]
i am having the same issue with new version of smart plug with power monitoring older devices worked.  
https://www.aliexpress.com/item/1005005167780031.html?spm=a2g0o.order_list.order_list_main.59.6d1a18023TMPJc
above is link to what i bought, even forcing doesnt help. if i manually enter the ip and try to get power i still cant get power usage stats

**Comment by demonreeper** [comment]
![Screenshot 2023-08-07 024013](https://github.com/jasonacox/tinytuya/assets/4169360/441126b8-3357-4482-8c0c-a985d2c71a28)
![Screenshot 2023-08-07 023758](https://github.com/jasonacox/tinytuya/assets/4169360/44ba424c-afff-4223-87b8-e3666b0ebcc0)
i am able to cloud connect and get the data through it using .cloud

**Reply by jasonacox** [reply]
Interesting. I wonder if these are devices with a new firmware version that doesn't broadcast UDP discovery packets the same way or just has issues on the local network. 

I see you did a force scan and it identified two devices that it couldn't identify: 192.168.86.183 and 184. Could these be the devices?

<img width="601" alt="image" src="https://github.com/jasonacox/tinytuya/assets/836718/0944f62c-9459-4c8b-91f5-b810d0f606ee">

You could try debug mode as well:

```bash
python -m tinytuya scan -force 192.168.x.y/24 -d
```

**Reply by demonreeper** [reply]
Correct 183 and 184 are the devices.  i have reserved them ips. I was also able to get device ids and local keys from raw json file. I will try debug mode when i am back at home. anything else you need these are new devices they even appear slightly different on the tuya smart app.

**Reply by uzlonewolf** [reply]
What version of TinyTuya do you have?  `python3 -m tinytuya -h` should report it.

**Comment by demonreeper** [comment]
its latest i updated to v1.12.9 and tried, also tried from another windows machine was the same

**Comment by demonreeper** [comment]
here is debug log hope i masked all that i need to --##maskedmasked## this is what i have replaced where its empty it was empty


DEBUG:TinyTuya [1.12.9]

DEBUG:Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)] on win32
DEBUG:Using PyCrypto (3, 18, '0')
DEBUG:loaded=devices.json [5 devices]

TinyTuya (Tuya device scanner) [1.12.9]

[Loaded devices.json - 5 devices]

Scanning on UDP ports 6666 and 6667 and 7000 for devices for 18 seconds...

DEBUG:Listening for Tuya devices on UDP ports 6666, 6667 and 7000
    Option: Network force scanning requested.


    Running Scan...
DEBUG:Received valid UDP packet: {'ip': '192.168.86.185', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjnuy4s3kre7m7', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa0000000100000003000000449c7c27fbcea3c0079a055de50471a819c777f161e4f98f11874c5a98fcfd4d10aed88f0dcce733a3c50b1abebe9a370d3c4a44f0ebf6f42b099c15432cc444450000aa55'
DEBUG:decrypting=b'9\xba\x1e^V\x13\xe2\xbc\x8e\xed\tT)7\x13m@g\xf3\x05\xd5\x85-\x93\xe9C\x1a\x81\xc7\xf54E9\x91\xf2\xb3\xd6\xf7\xbaA\xf4\xdb\x9cE5\xa1\x95\xaf\xc7w\xf1a\xe4\xf9\x8f\x11\x87LZ\x98\xfc\xfdM\x10'
DEBUG:decrypted session key negotiation step 2 payload=b'240d8d8c6a0c969c\x88\xe0\xe1\xdd\xa5\x83\xd6N?\x10P~QL+h\xb6\xc4\xe3c!\x90\xa1\xac<\xc2\xff\xcb\x9f$\x103'
DEBUG:payload type = <class 'bytes'> len = 48
DEBUG:session local nonce: b'0123456789abcdef' remote nonce: b'240d8d8c6a0c969c'
DEBUG:final payload: b't\xdd\x88H^\xf0)s\x00Kt\x1f\xa2\x0e\x13.\x8e\x05\xef\x1eH\xfd\x8c\xf5*\xb7\xb2\xcd\xa3gL\xc9'
DEBUG:payload encrypted=b'000055aa0000000200000005000000542d5f18319b9180346c6380f54407adb8e838272fa8d50767afbfc171d86e61e3c777f161e4f98f11874c5a98fcfd4d101c861f1aff20ea8d4d2cec7f48b5bf33396294a52eab2c576bf89eeb01ba59e90000aa55'
DEBUG:Session nonce XOR'd: b'\x02\x05\x02W\x0cQ\x0eT\x0eXQ\x01ZR\\\x05'
DEBUG:Session key negotiate success! session key: b'\x94\x83\x10QzK"\x85l\xba]\x7f8\x81\xd2!'
DEBUG:final payload_dict for '##maskedmasked##' ('v3.4'/'default'): {1: {'command': {'gwId': '', 'devId': '', 'uid': '', 't': ''}}, 7: {'command': {'protocol': 5, 't': 'int', 'data': {}}, 'command_override': 13}, 8: {'command': {'gwId': '', 'devId': ''}}, 9: {'command': {'gwId': '', 'devId': ''}}, 10: {'command': {}, 'command_override': 16}, 13: {'command': {'protocol': 5, 't': 'int', 'data': {}}}, 16: {'command': {}}, 18: {'command': {'dpId': [18, 19, 20]}}, 64: {'command': {'reqType': '', 'data': {}}}}
DEBUG:building command 10 payload=b'{}'
DEBUG:final payload: b'{}'
DEBUG:payload encrypted=b'000055aa0000000300000010000000349affdb329e8197e6aaced645fb94adf2a641041102cc3e4bff04adf8af460e75671e4db08b6b66fb325b74422fff74430000aa55'
DEBUG:PollDevice: raw unpacked message = TuyaMessage(seqno=51428, cmd=16, retcode=0, payload=b'\xd5~\xce\xd2\x93\x8a\xf2\xef\xbd\xc1s\xd1\x13x\x0e\xc4\xce\x7f\x1f\x82\x15\xa8\x92\xaf\xc7\xbd\xb4l\x1aB\x03\xd4\x98\xdb^\x89\xaaR\x8d!\xa8:\x88<i\x82\x126\xd1\xf7q\\\xe7R\xa81\xf8\xf6*3,\x00b\xa1Rx[\xbd\x01(J\x97\x1a\xd3e\r\xf2d\xe0\x8f', crc=b"\xd7\\\x0c\xf1\xf5[;\x05\xd3\x92\xae4|\xa3'W\x8b\x9bht\xf7Hx\x9b\xcf\xb9x\xe5\x85k\xe7`", crc_good=True, prefix=21930, iv=None)
DEBUG:decode payload=b'\xd5~\xce\xd2\x93\x8a\xf2\xef\xbd\xc1s\xd1\x13x\x0e\xc4\xce\x7f\x1f\x82\x15\xa8\x92\xaf\xc7\xbd\xb4l\x1aB\x03\xd4\x98\xdb^\x89\xaaR\x8d!\xa8:\x88<i\x82\x126\xd1\xf7q\\\xe7R\xa81\xf8\xf6*3,\x00b\xa1Rx[\xbd\x01(J\x97\x1a\xd3e\r\xf2d\xe0\x8f'
DEBUG:decrypting=b'\xd5~\xce\xd2\x93\x8a\xf2\xef\xbd\xc1s\xd1\x13x\x0e\xc4\xce\x7f\x1f\x82\x15\xa8\x92\xaf\xc7\xbd\xb4l\x1aB\x03\xd4\x98\xdb^\x89\xaaR\x8d!\xa8:\x88<i\x82\x126\xd1\xf7q\\\xe7R\xa81\xf8\xf6*3,\x00b\xa1Rx[\xbd\x01(J\x97\x1a\xd3e\r\xf2d\xe0\x8f'
DEBUG:decrypted 3.x payload=b'{"dps":{"1":false,"9":0,"38":"off","42":"","43":"","44":"","47":"flip"}}'
DEBUG:payload type = <class 'bytes'>
DEBUG:decoded results='{"dps":{"1":false,"9":0,"38":"off","42":"","43":"","44":"","47":"flip"}}'
Dressing Light   Product ID = keyjnuy4s3kre7m7  [Valid Broadcast]:
    Address = 192.168.86.185   Device ID = ##maskedmasked## (len:22)  Local Key = ##maskedmasked##  Version = 3.4
 Type = default, MAC = 1c:90:ff:f7:d9:1d
    Status: {'1': False, '9': 0, '38': 'off', '42': '', '43': '', '44': '', '47': 'flip'}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.250', 'gwId': '##maskedmasked##', 'active': 2, 'ability': 0, 'mode': 0, 'encrypt': True, 'productKey': 'ymf4oruxqx0xlogp', 'version': '3.3'}
Unknown v3.3 Device   Product ID = ymf4oruxqx0xlogp  [Valid Broadcast]:
    Address = 192.168.86.250   Device ID = ##maskedmasked## (len:20)  Local Key =   Version = 3.3  Type = default, MAC = a4:cf:12:cb:34:cf
    No Stats for 192.168.86.250: DEVICE KEY required to poll for status
DEBUG:Received valid UDP packet: {'ip': '192.168.86.182', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.180', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:final payload_dict for '##maskedmasked##' ('v3.3'/'default'): {1: {'command': {'gwId': '', 'devId': '', 'uid': '', 't': ''}}, 7: {'command': {'devId': '', 'uid': '', 't': ''}}, 8: {'command': {'gwId': '', 'devId': ''}}, 9: {'command': {'gwId': '', 'devId': ''}}, 10: {'command': {'gwId': '', 'devId': '', 'uid': '', 't': ''}}, 13: {'command': {'devId': '', 'uid': '', 't': ''}}, 16: {'command': {'devId': '', 'uid': '', 't': ''}}, 18: {'command': {'dpId': [18, 19, 20]}}, 64: {'command': {'reqType': '', 'data': {}}}}
DEBUG:building command 10 payload=b'{"gwId":"##maskedmasked##","devId":"##maskedmasked##","uid":"##maskedmasked##","t":"1691432184"}'
DEBUG:payload encrypted=b'000055aa000000010000000a000000888f6b0d8dc529eeabff10416778ceb5f22233de8e60a2a088b4296ed6ce1afca05c4d07c791d6a78ed4b488d06102e84783f1df04eadbe85dd40446122dceb1c02820d0c986ece3d756e684d7c22930412233de8e60a2a088b4296ed6ce1afca0ce58d7e528ca295ff2951809908b3eb91d7447a5e153aafa899ac0a2d41ca3f26eab464d0000aa55'
DEBUG:PollDevice: raw unpacked message = TuyaMessage(seqno=1, cmd=10, retcode=0, payload=b'\x843O(s\x0c\x9f\x17\xcbN\x99\x12\xd3\xd26M\xba@HKn\xfeU\xff\xb1\xeb\xb2+\x90\x9c\xa8\x86\xf8\xc8\x7f\x84V\xe2\xa1\x96LXT\x14_\x9c6r\x04;\xdf\x1f\xa7\xa3\xed\x8fy\x07\xc3\xc8\x95G0m#m\xe8\xc8\x11\xb4, \x01\xbb\x03\xd7\x17\xf3\xc1\x9d\x08i\x84K$\xcb\xc8\x92\ry\xec\xc32\xa2i\xc9<"\x89\xa9v \xa0/\x00\xa8\xb9\x81\x16\xda\x1b\x93:O{\xbf\xd7Ay\xa7Z\x0fF\xc8j\xaa\x14\x16)\xee\xfc\x0c(8Oc\x8fKSh\x80\x13\x1a\x8d8<\xb5\xe3f=\x0c\x80\xd3\x18\xf3\xa9vgG[\xe1\xd0\x9eCFB\xb7\x9d?"\xab\x88#\xec=\xe4\xd4s\xa6\xe0\xcf\xd4^s\x99-\xfcb\xce\x0e@1', crc=2898534991, crc_good=True, prefix=21930, iv=None)
DEBUG:decode payload=b'\x843O(s\x0c\x9f\x17\xcbN\x99\x12\xd3\xd26M\xba@HKn\xfeU\xff\xb1\xeb\xb2+\x90\x9c\xa8\x86\xf8\xc8\x7f\x84V\xe2\xa1\x96LXT\x14_\x9c6r\x04;\xdf\x1f\xa7\xa3\xed\x8fy\x07\xc3\xc8\x95G0m#m\xe8\xc8\x11\xb4, \x01\xbb\x03\xd7\x17\xf3\xc1\x9d\x08i\x84K$\xcb\xc8\x92\ry\xec\xc32\xa2i\xc9<"\x89\xa9v \xa0/\x00\xa8\xb9\x81\x16\xda\x1b\x93:O{\xbf\xd7Ay\xa7Z\x0fF\xc8j\xaa\x14\x16)\xee\xfc\x0c(8Oc\x8fKSh\x80\x13\x1a\x8d8<\xb5\xe3f=\x0c\x80\xd3\x18\xf3\xa9vgG[\xe1\xd0\x9eCFB\xb7\x9d?"\xab\x88#\xec=\xe4\xd4s\xa6\xe0\xcf\xd4^s\x99-\xfcb\xce\x0e@1'
DEBUG:decrypting=b'\x843O(s\x0c\x9f\x17\xcbN\x99\x12\xd3\xd26M\xba@HKn\xfeU\xff\xb1\xeb\xb2+\x90\x9c\xa8\x86\xf8\xc8\x7f\x84V\xe2\xa1\x96LXT\x14_\x9c6r\x04;\xdf\x1f\xa7\xa3\xed\x8fy\x07\xc3\xc8\x95G0m#m\xe8\xc8\x11\xb4, \x01\xbb\x03\xd7\x17\xf3\xc1\x9d\x08i\x84K$\xcb\xc8\x92\ry\xec\xc32\xa2i\xc9<"\x89\xa9v \xa0/\x00\xa8\xb9\x81\x16\xda\x1b\x93:O{\xbf\xd7Ay\xa7Z\x0fF\xc8j\xaa\x14\x16)\xee\xfc\x0c(8Oc\x8fKSh\x80\x13\x1a\x8d8<\xb5\xe3f=\x0c\x80\xd3\x18\xf3\xa9vgG[\xe1\xd0\x9eCFB\xb7\x9d?"\xab\x88#\xec=\xe4\xd4s\xa6\xe0\xcf\xd4^s\x99-\xfcb\xce\x0e@1'
DEBUG:decrypted 3.x payload='{"dps":{"1":true,"9":0,"17":4,"18":95,"19":81,"20":2283,"21":1,"22":627,"23":26684,"24":15227,"25":2740,"26":0,"38":"memory","39":false,"40":"none","41":false,"42":"","43":"","44":""}}'
DEBUG:payload type = <class 'str'>
DEBUG:decoded results='{"dps":{"1":true,"9":0,"17":4,"18":95,"19":81,"20":2283,"21":1,"22":627,"23":26684,"24":15227,"25":2740,"26":0,"38":"memory","39":false,"40":"none","41":false,"42":"","43":"","44":""}}'
Smart  Plug TV   Product ID = keyjup78v54myhan  [Valid Broadcast]:
    Address = 192.168.86.180   Device ID = ##maskedmasked## (len:22)  Local Key = ##maskedmasked##  Version = 3.3
 Type = default, MAC = 38:1f:8d:cc:7c:f9
    Status: {'1': True, '9': 0, '17': 4, '18': 95, '19': 81, '20': 2283, '21': 1, '22': 627, '23': 26684, '24': 15227, '25': 2740, '26': 0, '38': 'memory', '39': False, '40': 'none', '41': False, '42': '', '43': '', '44': ''}
DEBUG:Starting brute force network scan 192.168.86.0/24
  Starting Scan for network 192.168.86.0/24
DEBUG:Received valid UDP packet: {'ip': '192.168.86.185', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjnuy4s3kre7m7', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.250', 'gwId': '##maskedmasked##', 'active': 2, 'ability': 0, 'mode': 0, 'encrypt': True, 'productKey': 'ymf4oruxqx0xlogp', 'version': '3.3'}
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:payload encrypted=b'000055aa000000010000000a00000008354862b10000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:payload encrypted=b'000055aa000000010000000a00000008354862b10000aa55'
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:payload encrypted=b'000055aa000000010000000a00000028deadbeef112233445566778899aabbccddeeffb00bface112233feedbabe74f0149a707c0000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:payload encrypted=b'000055aa000000010000000a00000028deadbeef112233445566778899aabbccddeeffb00bface112233feedbabe74f0149a707c0000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.182', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.180', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.185', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjnuy4s3kre7m7', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.250', 'gwId': '##maskedmasked##', 'active': 2, 'ability': 0, 'mode': 0, 'encrypt': True, 'productKey': 'ymf4oruxqx0xlogp', 'version': '3.3'}
Smart Plug PC   Product ID = keyjup78v54myhan  [Valid Broadcast]:
    Address = 192.168.86.182   Device ID = ##maskedmasked## (len:22)  Local Key = ##maskedmasked##  Version = 3.3
 Type = default, MAC = d8:1f:12:88:93:b8
    Polling 192.168.86.182 Failed:
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:payload encrypted=b'000055aa000000010000000a00000028deadbeef112233445566778899aabbccddeeffb00bface112233feedbabe74f0149a707c0000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:payload encrypted=b'000055aa000000010000000a00000028deadbeef112233445566778899aabbccddeeffb00bface112233feedbabe74f0149a707c0000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.182', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.180', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:payload encrypted=b'000055aa000000010000000d000000088768bea10000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:payload encrypted=b'000055aa000000010000000d000000088768bea10000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.185', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjnuy4s3kre7m7', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.250', 'gwId': '##maskedmasked##', 'active': 2, 'ability': 0, 'mode': 0, 'encrypt': True, 'productKey': 'ymf4oruxqx0xlogp', 'version': '3.3'}
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:payload encrypted=b'000055aa000000010000000d00000028deadbeef112233445566778899aabbccddeeffb00bface112233feedbabe74f020a819a30000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:payload encrypted=b'000055aa000000010000000d00000028deadbeef112233445566778899aabbccddeeffb00bface112233feedbabe74f020a819a30000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.182', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004412428dd82939a7a0b582fce8a3987583fd7a9f9b249a5dc04403bbe572362764da1673c8e72b039ba3d41cf8ca6fc7f98ace4c4630b79a73ec8b6c455bbe5af60000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004412428dd82939a7a0b582fce8a3987583fd7a9f9b249a5dc04403bbe572362764da1673c8e72b039ba3d41cf8ca6fc7f98ace4c4630b79a73ec8b6c455bbe5af60000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.180', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
DEBUG:Received valid UDP packet: {'ip': '192.168.86.185', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjnuy4s3kre7m7', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004412428dd82939a7a0b582fce8a3987583fd7a9f9b249a5dc04403bbe572362764da1673c8e72b039ba3d41cf8ca6fc7f98ace4c4630b79a73ec8b6c455bbe5af60000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004412428dd82939a7a0b582fce8a3987583fd7a9f9b249a5dc04403bbe572362764da1673c8e72b039ba3d41cf8ca6fc7f98ace4c4630b79a73ec8b6c455bbe5af60000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.250', 'gwId': '##maskedmasked##', 'active': 2, 'ability': 0, 'mode': 0, 'encrypt': True, 'productKey': 'ymf4oruxqx0xlogp', 'version': '3.3'}
DEBUG:Force-Scan Found Device 192.168.86.183
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004412428dd82939a7a0b582fce8a3987583fd7a9f9b249a5dc04403bbe572362764da1673c8e72b039ba3d41cf8ca6fc7f98ace4c4630b79a73ec8b6c455bbe5af60000aa55'
DEBUG:Force-Scan Found Device 192.168.86.184
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa00000001000000030000004412428dd82939a7a0b582fce8a3987583fd7a9f9b249a5dc04403bbe572362764da1673c8e72b039ba3d41cf8ca6fc7f98ace4c4630b79a73ec8b6c455bbe5af60000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.86.182', 'gwId': '##maskedmasked##', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': 'keyjup78v54myhan', 'version': '3.3'}
Unknown v?? Device   Product ID = ?  [Failed to Force-Scan]:
    Address = 192.168.86.183   Device ID =  (len:0)  Local Key =   Version = ??  Type = default, MAC =
    Polling 192.168.86.183 Failed: Device stopped responding before key was found
Unknown v?? Device   Product ID = ?  [Failed to Force-Scan]:
    Address = 192.168.86.184   Device ID =  (len:0)  Local Key =   Version = ??  Type = default, MAC =
    Polling 192.168.86.184 Failed: Device stopped responding before key was found
Scan completed in 23.8134 seconds

Scan Complete!  Found 6 devices.
Broadcasted: 4
Force-Scanned: 2  - Matched GWID: 0 Matched Key: 0 Unmatched: 2
Force-Scan Errors: Connection Errors: 2 Version Detect Failed: 0
Versions: 0.0: 2, 3.3: 3, 3.4: 1
Unknown Devices: 3

>> Saving device snapshot data to snapshot.json

DEBUG:Scan complete with 6 devices found

**Reply by uzlonewolf** [reply]
It looks to me like the device isn't broadcasting and the brute-force version detection in the scanner isn't working.  Since it is accepting connections I suspect it will work if you pass the correct version to the `d = tinytuya.Device(...)` call.  I'd try 3.5, 3.4, and 3.3 in that order.

**Reply by demonreeper** [reply]
![image](https://github.com/jasonacox/tinytuya/assets/4169360/06075b83-d455-4372-8920-42a4242866f7)
get the same error with all 3 versions

**Reply by demonreeper** [reply]
![image](https://github.com/jasonacox/tinytuya/assets/4169360/b027270c-4564-49d1-93a2-cab89c0d8ba6)
this is using force method and updatedps

**Comment by demonreeper** [comment]
https://pastebin.com/QWzijAKe

**Comment by rnlgreen** [comment]
I'm seeing exactly the same thing with newer firmware plugs. Thinking I might have to write my own code to supplement the json with fixed IPs, unless there's a chance this will get fixed?

**Reply by uzlonewolf** [reply]
Fix what?  If device broadcasts aren't being sent/received and the device isn't responding to force-scans then there really isn't anything we can do in TinyTuya.  You're going to need to contact the device manufacturer and see if you can get them to fix their firmware.

At this point I suspect it is an incompatibility between the device and the WiFi router.  I bought both of the devices linked to by @demonreeper (as well as a 3rd in a different listing from the same seller) and all 3 have been working fine for me without issue for over a month; I'm running a UniFi UAP-AC-Pro access point (running an older firmware as all the newer ones are horribly broken for IoT devices) and a generic Linux box handling DHCP and gateway/router duties.

**Reply by rnlgreen** [reply]
Apologies, perhaps I should have said "resolved" rather than fixed; wasn't trying to suggest tinytuya was at fault!

I've actually realised it was user error on my part; I use Node-RED and it seems my newer plugs ("Defender Mini Plug") don't broadcast or accept secondary connections unlike my earlier Teckin SP27 plugs which both broadcast and accept additional connections.

Thanks for the great software, I'm just going to have to switch a few plugs around so I can do what I need with the ones that play nicely.

**Reply by uzlonewolf** [reply]
I'm running 6.2.49.14111 on everything as I've found 6.4.x and 6.5.x to all have major issues with devices dropping offline for no apparent reason.  I've seen posts in Ubnt's forums and on Reddit saying others are seeing the exact same issue, so it's not just me.  The UAP-AC-Pro and U6-LR also seem to be a bit more reliable for IoT devices than the UAP6MP.

Do you have a link for those plugs?  I'm always interested in picking up "problematic" devices to poke at.

**Reply by rnlgreen** [reply]
I got them off Amazon in the UK: [Defender Mini Smart Plug](https://www.amazon.co.uk/Smart-Plug-Alexa-Google-Home/dp/B08LDVLH6R?pd_rd_w=dIpNE&content-id=amzn1.sym.ae973372-04ab-4f43-8bdd-219dc0aabb13&pf_rd_p=ae973372-04ab-4f43-8bdd-219dc0aabb13&pf_rd_r=BCEB42Q1M1MY35FGQ3G0&pd_rd_wg=zJAIv&pd_rd_r=66136ae5-4e3f-4f98-b4ca-a1f648b17f14&pd_rd_i=B08LDVLH6R&ref_=pd_bap_d_grid_rp_0_19_i&th=1)

They are physically bigger than the Teckin plugs I've used previously, so don't fit in a four-way extension lead, and as I put in my updated post they just don't broadcast or support second connections when my Node-RED is connected to them, which is frustrating but I can probably get round it.

Interesting on the Unifi firmware. Now you mention it, the Teckin plugs do have a habit of losing their minds once in a while (flashing orange) and I need to readopt them, but I don't know if that's caused by Wi-Fi issues, I'll have to try downgrading one of my APs and pinning the plugs to that.

**Comment by biggiebk** [comment]
I don't suppose anybody has found a solution to this?  Just picked up a switch that has the behavior.  No IP or Version is found.

```json
    {
        "name": "BR Switch",
        "id": "ebb1a838262de95b89glbe",
        "key": "--------",
        "mac": "a8:80:55:84:73:5b",
        "uuid": "956cb19ca1b75dee",
        "sn": "100103513001B0",
        "category": "kg",
        "product_name": "WIFI Smart Switch",
        "product_id": "dn7fpj6lcz3krzbv",
        "biz_type": 18,
        "model": "NH-\u7f8e\u5f0f\u5355\u706b\u6309\u952e\u4e00\u8def",
        "sub": false,
        "icon": "https://images.tuyaus.com/smart/icon/ay1554975589161DYAhk/76f66b4c3a3ce2567c60ee4f736ae57d.png",
        "mapping": {
            "1": {
                "code": "switch_1",
                "type": "Boolean",
                "values": {}
            },
            "7": {
                "code": "countdown_1",
                "type": "Integer",
                "values": {
                    "unit": "s",
                    "min": 0,
                    "max": 86400,
                    "scale": 0,
                    "step": 1
                }
            },
            "14": {
                "code": "relay_status",
                "type": "Enum",
                "values": {
                    "range": [
                        "power_off",
                        "power_on",
                        "last"
                    ]
                }
            },
            "15": {
                "code": "light_mode",
                "type": "Enum",
                "values": {
                    "range": [
                        "none",
                        "relay",
                        "pos"
                    ]
                }
            }
        },
        "ip": "",
        "version": ""
    }
```

**Reply by uzlonewolf** [reply]
You can try `python3 -m tinytuya scan -force` to see if it helps.

**Reply by biggiebk** [reply]
Yeah unfortunately that does not seem to be doing the trick for me.  I've pinged the manufacturer and will cross my fingers that they will care enough to do something about it, but I have my doubts.
